### PR TITLE
Get statuses from api

### LINKED ISSUE #518: Use code-gov-api to get status data
Your issue may already be reported!
Please search on the [issue track](../) before creating one.

## Expected Behavior

Compliance dashboard should use the [status endpoint](https://api.code.gov/status.json) found in [code-gov-api](https://github.com/GSA/code-gov-api).

## Current Behavior

Compliance dashboard is consuming the report.json found in its assets folder.

## Possible Solution

Change the compliance dashboard component to use the api client component to get it's data.

## Context

This would create some confusion on what the source of truth is for our data.

### PR DESCRIPTION
**Pull request by froi**
**Summary**

Change compliance dashboard to use the API instead of a static file.

This PR fixes/implements the following **bugs/features**

* [x] #518

Explain the **motivation** for making this change. What existing problem does the pull request solve?

All endpoints of the API should be implemented. The API is the source of truth for our data and we should not be using static files anymore.

**Test plan (required)**

A manual/visual test was done from localhost:

image

The selected asset in the network tab of the developer tools shows that `status.json` is being requested from `https://code-api.app.cloud.gov/api/status.json`.

In contrast our current production build uses the static report.json:

image

**Closing issues**

Closes #518

### CODE REVIEW

**Comment by DanielJDufour** [comment]
@froi , is this WIP? Want me to review?

File: src/app/components/policy-guide/docs/compliance/compliance-dashboard/compliance-dashboard.component.ts [hunk 1]
```diff
@@ -43,58 +43,64 @@ export class ComplianceDashboardComponent implements OnInit, OnDestroy {
     );
   }
 
+  _setRequirementStatuses(agencyRequirements) {
+    let requirements = [];
+    let overallStatus = 'noncompliant';
+
+    for (let requirement in agencyRequirements) {
+      if (agencyRequirements.hasOwnProperty(requirement)) {
```
[the remark this answers is not in the document]
Comment (froi) [reply src/app/components/policy-guide/docs/compliance/compliance-dashboard/compliance-dashboard.component.ts]: I agree but the TS linter screams at me if the `hasOwnProperty` isn't done.

**Comment by DanielJDufour** [comment]
Looks good, but a couple test are failing because API Key isn't set to work on circleci.  I'll probably merge this on Sunday night or Monday, but also need to look at creating and using API_KEY env variable on circle

**Comment by DanielJDufour** [comment]
Previous PR: Latest commit 2d1d506  6 days ago

**Pull request merged**